### DISCUSSION TITLE
Pre-compiled binaries

### DISCUSSION THREAD

**Discussion by alexmozaidze** [flxzt/rnote#778] [Ideas]
Having pre-built binaries as part of releases would be a nice addition, especially for people with low-end machines. Flat package is extremely large and attempts to compile the project via `cargo install --git https://github.com/flxzt/rnote rnote` were futile, since I didn't have enough memory.

In short: Can you publish compiled binary for x64 Linux and Windows?

**Comment by flxzt** [comment]
This won't work, the application is not self-contained and requires installing some files to the system/environment. It also is dependent on some dynamically linked libraries (poppler, gtk, ..).

I would suggest to you to try the flatpak again: If you don't count the flatpak runtime which is shared between applications, the app itself is not really that big (~45MB).

**Reply by alexmozaidze** [reply]
I need to install 1GB worth of Gnome environment in order to install rnote, which is too much considering my limited disk space.

Maybe you could make an AppImage and see if it's <=200MB? An AppImage should typically be smaller in size compared to flatpak package.

**Reply by flxzt** [reply]
I personally am committed to distributing the application through flatpak. Also: If you are building the project manually, you would need to build/install it through meson as outlined in [BUILDING.md](https://github.com/flxzt/rnote/blob/main/BUILDING.md#build-with-meson)

**Comment by fiftydinar** [comment]
@alexmozaidze

I don't know if you are still around here, but for everyone who has the same concerns as you,

I maintain Rnote AppImage, which is 45.4MB in size as of now, bundling everything the application needs, without depending on libraries from the host's system (only `sh` being available, which every Linux has). Sounded impossible to @flxzt back then, might sound impossible now, but it's fortunately possible. For more details about how that's achieved, you can look at this and the AnyLinux-AppImages repo:

https://github.com/pkgforge-dev/Rnote-AppImage

https://github.com/pkgforge-dev/AnyLinux-AppImages
